components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -154,14 +163,56 @@ private void logOperationExecutionResults(ActionType actionType,
+                                                        Event actionEvent,
+                                                        ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        processServerError(failureResponse.getFailureReason());
+        return new FailedStatus(new Failure(failureResponse.getFailureReason(),
+                failureResponse.getFailureDescription()));
+    }
+
+    private void processServerError(String errorCode)
+            throws ActionExecutionResponseProcessorException {
+
+        if (isServerError(errorCode)) {
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                        ActionExecutionLogConstants.ACTION_EXECUTION_COMPONENT_ID,
+                        ActionExecutionLogConstants.ActionIDs.EXECUTE_ACTION_OPERATIONS);
+                diagLogBuilder
+                        .resultMessage("An exception was thrown for action. FAILED status should not be used to " +
+                                "process server errors.")
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                        .build();
+                LoggerUtils.triggerDiagnosticLogEvent(diagLogBuilder);
+            }
+            throw new ActionExecutionResponseProcessorException("FAILED status should not be used to process" +
+                    " server errors.");
+        }
+    }
+
+    private boolean isServerError(String errorCode) {
+
+        return (errorCode.equalsIgnoreCase(ServerErrorCode.INTERNAL_SERVER_ERROR.code.trim()) ||
+                errorCode.equalsIgnoreCase(ServerErrorCode.SERVER_ERROR.code.trim()) ||
+                errorCode.equalsIgnoreCase(ServerErrorCode.ERROR_500.toString()));
+    }
+
+    @Override
+    public ActionExecutionStatus<Error> processErrorResponse(Map<String, Object> map, Event event,
                                                       ActionInvocationErrorResponse actionInvocationErrorResponse)
             throws ActionExecutionResponseProcessorException {
 
-        //todo: need to implement to process the error so that if a processable error is received
-        // it is communicated to the client.
-        // we will look into this as we go along with other extension types validating the way to model this.
-        return null;
+        // Client and server errors that occur when calling the service implementing the extension are reported
+        // as Internal_Server_Error.
+        // The error description could be utilized to offer additional context by passing along the
+        // original error returned by the service implementing the extension.
+        // However, currently this value is not propagated by the endpoint to comply with OAuth specification.
```

Comment:
As this is long use block comment format rather than line comment format

Reply:
This has been fixed.